### PR TITLE
Add LLM::File

### PR DESCRIPTION
**Pull request by 0x1eef**
Gemini support only (at least for now)

Example:
```ruby
require "llm"
llm = LLM.gemini(key)
bot = llm.chat LLM::File("/path/to/image.png")
bot.chat "Describe the image"
bot.messages.each do |message|
  print message.role, ": ", message.content.to_s, "\n"
end
```

### CODE REVIEW

**Comment by 0x1eef** [comment]
@antaz This pull request also includes [a bug fix](https://github.com/antaz/llm/pull/55/commits/4f937a1635310b1cbf56cdad5ad067e77b535035). It looks like we didn't catch every situation when updating the `complete` method recently

File: README.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ end
 
 ##
 # [system] keep the answer concise
-# [user] [{:type=>:image_url, :image_url=>{:url=>"https://upload.wikimedia.org/wikipedia/commons/b/be/Red_eyed_tree_frog_edit2.jpg"}}]
+# [user] https://upload.wikimedia.org/wikipedia/commons/b/be/Red_eyed_tree_frog_edit2.jpg
```
Comment (antaz) [inline README.md:59]: Maybe we should specify whether this was sent as a URI/File or just text?

File: lib/llm/providers/gemini.rb [hunk 2]
```diff
@@ -29,21 +29,37 @@ def embed(input, **params)
...
+      if LLM::File === prompt
+        file = prompt
+        inline_data = {mime_type: file.mime_type, data: file.to_base64}
+        {inline_data:}
+      elsif String === prompt
+        {text: prompt}
+      else
+        raise TypeError, "#{self.class} does not support #{prompt.class} objects"
+      end
+    end
```
Comment (antaz) [inline lib/llm/providers/gemini.rb:61]: This is good, but i think we should also support passing data as URI directly (using `file_data` and `file_uri`) and leave File for local files?
[https://ai.google.dev/gemini-api/docs/vision?lang=rest](url)

**Review by antaz — COMMENTED** [review]
This is good overall. Left some comments.

File: lib/llm/providers/gemini.rb (hunk 2 above)
Comment (antaz) [reply lib/llm/providers/gemini.rb:61]: > This is good, but i think we should also support passing data as URI directly (using `file_data` and `file_uri`) and leave File for local files? [https://ai.google.dev/gemini-api/docs/vision?lang=rest](url)

i.e if `File` then inline through base64 and if `URI` pass it directly with `file_uri`

Comment (0x1eef) [reply lib/llm/providers/gemini.rb:61]: > This is good, but i think we should also support passing data as URI directly (using file_data and file_uri) and leave File for local files?

I think that's a separate pull request. Gemini didn't support `URI` to begin with. It should be added separately.

File: README.md (hunk 1 above)
Comment (0x1eef) [reply README.md:59]: Not in scope for this PR. But agreed, we should handle more than "text" and that goes for the response level too.

File: lib/llm/providers/gemini.rb (hunk 2 above)
Comment (antaz) [reply lib/llm/providers/gemini.rb:61]: Yeah. You are right! I was thinking of this PR as a vision support for Gemini :sweat_smile:

**Pull request merged**